DistantWorlds2.ModLoader.Core/GameDataDefinitionPatching.cs

```diff
@@ -136,17 +146,148 @@ public static class GameDataDefinitionPatching
     public static readonly MethodInfo MiGenericPatchDefinitions
         = typeof(GameDataDefinitionPatching).GetMethod(nameof(GenericPatchDefinitions))!;
 
+    private static readonly Dictionary<Type, Action<object>> RebuildIndexMethods = new Dictionary<Type, Action<object>>();
+    private static readonly Dictionary<Type, bool> IsIndexedListCache = new Dictionary<Type, bool>();
 
     public static readonly MmVariableDsl Dsl = new();
 
-    public static void ApplyContentPatches(string dataPath, Galaxy galaxy)
+    public static void ApplyContentPatches()
+    {
+        foreach (var dataPath in ModLoader.ModManager.PatchedDataStack)
+        {
+            try
+            {
+                if (ModLoader.DebugMode)
+                    Console.WriteLine($"Applying content patches from {dataPath}");
+                ApplyContentPatches(dataPath);
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine($"Failure applying content patches from {dataPath}");
+                ModLoader.ModManager.OnUnhandledException(ExceptionDispatchInfo.Capture(ex));
+            }
+        }
+    }
+
+    private static bool IsIndexedList<T>(T list)
+    {
+        bool isIndexed;
+        if(IsIndexedListCache.TryGetValue(list.GetType(), out isIndexed))
+        {
+            return isIndexed;
+        }
+        var mi = list.GetType().GetMethod("RebuildIndexes", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
+        if (mi != null)
+        {
+            var parameter = Expression.Parameter(typeof(object));
+            RebuildIndexMethods.Add(list.GetType(), Expression.Lambda<Action<object>>(
+                Expression.Call(Expression.Convert(parameter, list.GetType()),mi), parameter)
+                .CompileFast());
+
+            IsIndexedListCache[list.GetType()] = true;
+            return true;
+        }
+        else
+        {
+            IsIndexedListCache[list.GetType()] = false;
+            return false;
+        }
```

Comment:
Can reduce 6 code instances of `list.GetType()` to a local var.
If `T` is the final type can use `typeof(T)` instead of `list.GetType()`.


N.B.